tox.ini

```diff
@@ -209,17 +208,34 @@ commands =
 
 description = upload coverage to Codecov
 
+depends = {[coverage_report]depends}
+
 basepython = python
 
 skip_install = True
 
 deps =
-    coverage==4.5.4
+    {[testenv:coverage_report]deps}
     codecov==2.0.22
 
-commands =
-    # Note documentation for CI variables in default environment's passenv
+passenv =
+    # See https://github.com/codecov/codecov-python/blob/master/README.md#using-tox
+    # And CI-specific docs:
+    #   https://help.github.com/en/articles/virtual-environments-for-github-actions#default-environment-variables
+    #   https://docs.travis-ci.com/user/environment-variables#default-environment-variables
+    #   https://www.appveyor.com/docs/environment-variables/
+    TOXENV CODECOV_* CI
+    GITHUB_*
+    TRAVIS TRAVIS_*
+    APPVEYOR APPVEYOR_*
 
+setenv =
+    {[testenv:coverage_report]setenv}
+
+    COVERAGE_XML={envlogdir}/coverage_report.xml
```

Comment:
This is moved down from the default environment, which doesn't need it, to here, which does.